main.cpp

```diff
@@ -238,14 +238,18 @@ IntType& IntType::operator*=(const int other)
 IntType& IntType::apply(std::function<IntType&(int&)> intFunc)
 {
     if(intFunc)
-    intFunc(*intTypeHeap);
+    {
+        intFunc(*intTypeHeap);
```

Comment:
`intFunc` returns `IntType&`.   That is the same thing this `apply` function returns. 
- [ ] return the result of calling the std::function if the std::function is valid.